src/main/webapp/WEB-INF/static/js/trialmanager/subobservations/subObservationSet.js

```diff
@@ -875,6 +841,83 @@
 				}
 			}
 
+			function applyNewValue() {
+				// setNewValue
+				var newValue = $scope.nested.newValueBatchUpdate;
+				if ($scope.nested.selectedVariableFilter.dataTypeCode === 'D') {
+					newValue = $.datepicker.formatDate("yymmdd", newValue);
+				}
+				var param = JSON.stringify({
+					newValue: newValue,
+					newCategoricalValueId: getCategoricalValueId($scope.nested.newValueBatchUpdate, $scope.nested.selectedVariableFilter),
+					observationUnitsSearchDTO: {
+						instanceId: $scope.nested.selectedEnvironment.instanceId,
+						draftMode: $scope.isPendingView,
+						filter: getFilter()
+					}
+				});
+				datasetService.setValueToVariable(subObservationSet.id, param).then(function () {
+					if ($scope.isPendingView) {
+						reloadDataset();
+					} else {
+						$scope.selectedStatusFilter = "1";
+						$scope.nested.selectedEnvironment = $scope.environments[1];
+						table().ajax.reload();
+						loadBatchActionCombo();
+					}
+				}, function (response) {
+					if (response.errors && response.errors.length) {
+						showErrorMessage('', response.errors[0].message);
+					} else {
+						showErrorMessage('', ajaxGenericErrorMsg);
+					}
+				});
+			}
+
+			function acceptValueAsIs() {
+				// acceptDraftDataByVariable
+				var param = JSON.stringify({
+					instanceId: $scope.nested.selectedEnvironment.instanceId,
+					draftMode: $scope.isPendingView,
+					filter: getFilter()
+				});
+				datasetService.acceptDraftDataByVariable(subObservationSet.id, param).then(function () {
+					reloadDataset();
+				}, function (response) {
+					if (response.errors && response.errors.length) {
+						showErrorMessage('', response.errors[0].message);
+					} else {
+						showErrorMessage('', ajaxGenericErrorMsg);
+					}
+				});
+			}
+
+			function deleteObservationsValues() {
+				const filter = getFilter();
+				filter.variableHasValue = true;
```

Comment:
is it necessary to set this in the frontend? why it cant be handled internally by the services? @mderamos